### PR TITLE
support multiple losses supervise the same prediction

### PR DESCRIPTION
**Pull request by Junjun2016**
support multiple losses supervise the same prediction

### CODE REVIEW

File: mmseg/models/decode_heads/decode_head.py [hunk 1]
```diff
@@ -33,7 +33,10 @@ class BaseDecodeHead(nn.Module, metaclass=ABCMeta):
                 a list and passed into decode head.
             None: Only one select feature map is allowed.
             Default: None.
-        loss_decode (dict): Config of decode loss.
+        loss_decode (dict|Sequence[dice]): Config of decode loss.
```
Comment (xvjiarui) [inline mmseg/models/decode_heads/decode_head.py:36]:
```suggestion
        loss_decode (dict | Sequence[dict]): Config of decode loss.
```

File: mmseg/models/decode_heads/decode_head.py [hunk 2]
```diff
@@ -69,9 +72,21 @@ def __init__(self,
...
+        if isinstance(loss_decode, dict):
+            self.loss_decode.append(build_loss(loss_decode))
+            self.loss_names.append('loss_seg')
+        elif isinstance(loss_decode, (list, tuple)):
+            for loss in loss_decode:
+                name = loss.pop('name')
+                self.loss_decode.append(build_loss(loss))
+                self.loss_names.append('loss_' + name)
+        else:
+            raise TypeError(f'loss_decode must be a dict or Sequence of dict,\
```
Comment (xvjiarui) [inline mmseg/models/decode_heads/decode_head.py:88]:
```suggestion
            raise TypeError(f'loss_decode must be a dict or sequence of dict,\
```

File: mmseg/models/decode_heads/decode_head.py [hunk 3]
```diff
@@ -224,10 +239,11 @@ def losses(self, seg_logit, seg_label):
         else:
             seg_weight = None
         seg_label = seg_label.squeeze(1)
-        loss['loss_seg'] = self.loss_decode(
-            seg_logit,
-            seg_label,
-            weight=seg_weight,
-            ignore_index=self.ignore_index)
+        for i in range(len(self.loss_decode)):
```
Comment (xvjiarui) [inline mmseg/models/decode_heads/decode_head.py:242]:
```suggestion
        for i in enumerate(self.loss_decode):
```

Comment (xvjiarui) [inline mmseg/models/decode_heads/decode_head.py:242]: Use `zip` instead.

**Comment by hellock** [comment]
Task linked: [CU-ggz63w multi-loss](https://app.clickup.com/t/ggz63w)
